test_runner/performance/test_storage_controller_scale.py

```diff
@@ -105,15 +106,24 @@ def test_storage_controller_many_tenants(
         ps.allowed_errors.append(".*request was dropped before completing.*")
 
     # Total tenants
-    tenant_count = 4000
+    small_tenant_count = 7900
+    large_tenant_count = 100
+    tenant_count = small_tenant_count + large_tenant_count
+    large_tenant_shard_count = 8
+    total_shards = small_tenant_count + large_tenant_count * large_tenant_shard_count
 
-    # Shards per tenant
-    shard_count = 2
-    stripe_size = 1024
+    # A small stripe size to encourage all shards to get some data
+    stripe_size = 1
 
-    total_shards = tenant_count * shard_count
+    tenants = list(TenantId.generate() for _i in range(0, tenant_count))
 
-    tenants = set(TenantId.generate() for _i in range(0, tenant_count))
+    # We will create timelines in only a subset of tenants, because creating timelines
+    # does many megabytes of IO, and we want to densely simulate huge tenant counts on
+    # a single test node.
+    tenant_timelines_count = 100
```

Comment:
`tenant_timelines_count` and `large_tenant_count` need to be equal for this to make sense. Would it be clearer if you merged them?

Reply:
Also, it would be nicer if we only had one list of tenant specifications. Each tenant is represented by a dict with tenant_id, optional timeline_id, shard_count and whatever else is needed.

Reply:
Refactored this a bit in 93f5ef3 so there is at least a Tenant object.  There are still a couple of lists of IDs though, because random.choice doesn't like operating on things like dict keys